### ISSUE TITLE
Dynamic backend generation is not working.

### ISSUE THREAD

**Issue by spielkind** [mittwald/kube-httpcache#55]
**Describe the bug**
Dynamic backend generation is not working

**To Reproduce**
1. use a large vcl template (>1MB , provided as volume)
2. respawn/recreate multiple (example 4 backends at once
3. old PODs will not be removed from backend
4. new PODs will not be added to backend

**Expected behavior**
kube-httpcache adds/removes backends

**Environment:**
 - Kubernetes version: 1.18
 - kube-httpcache version: [latest master]

**Logs**

```
panic: runtime error: index out of range [0] with length 0

goroutine 8 [running]:
github.com/mittwald/kube-httpcache/pkg/watcher.(*EndpointWatcher).watch(0xc0002e5a90, 0xc00004a480, 0xc00004a4e0)
	/workspace/pkg/watcher/endpoints_watch.go:93 +0xf1d
created by github.com/mittwald/kube-httpcache/pkg/watcher.(*EndpointWatcher).Run
	/workspace/pkg/watcher/endpoints_watch.go:17 +0x98

```

Or

```
W0219 16:27:58.037023       1 run.go:53] error while watching for updates: error while loading VCL (code 106): Message from C-compiler:
gcc: internal compiler error: Killed (program cc1)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-6/README.Bugs> for instructions.
Running C-compiler failed, exited with 4
VCL compilation failed
```

**Comment by martin-helmich** [comment]
Don't mind the bot, and for the noise -- I haven't found the time to investigate this, yet.

**Comment by martin-helmich** [comment]
The first issue (`panic: runtime error: index out of range [0] with length 0`) seems easy enough to fix, as there simply appears to be a missing slice length check before accessing `newBackendList` (in `endpoints_watch.go`, `endpoint.Subsets[0].Addresses` _is_ checked for length, but is filtered by readiness state _after that_, so `newBackendList` _might_ be empty if no backend pod is ready).

I'm still unsure about the VCL compiler error, though. Possibly a consequence of rendering the template with invalid data (like an empty backend list?)

**Comment by martin-helmich** [comment]
@spielkind I've added a preliminary fix in #64 that _should_ fix the runtime panic that you've encountered.

Regarding you second error: Can you test how your VCL template would render in case it was supplied with an empty backend list (which might happen, when no backend Pod is actually ready)? I'm suspecting the template might render to invalid VCL in this case, which then subsequently will trigger a compiler error when loading that template.
